### PR TITLE
NAS-104330 / 21.02 / Deprecate NIS support in TrueNAS Scale

### PR DESCRIPTION
**Pull request by anodos325**

### CODE REVIEW

File: src/middlewared/middlewared/etc_files/rc.conf.py [hunk 1]
```diff
@@ -237,8 +238,11 @@ def nfs_config(middleware, context):
 
 
 def nis_config(middleware, context):
+    if osc.IS_LINUX:
+        return []
+
     nis = middleware.call_sync('datastore.config', 'directoryservice.nis', {'prefix': 'nis_'})
-    if not nis['enable'] or not nis['domain']:
+    if nis['enable'] or not nis['domain']:
```
Comment (william-gr) [inline src/middlewared/middlewared/etc_files/rc.conf.py:245]: is this change intended?

File: src/middlewared/middlewared/plugins/directoryservices.py [hunk 2]
```diff
@@ -137,24 +151,31 @@ async def get_state(self):
...
             return ds_state
 
     @private
     async def set_state(self, new):
-        ds_state = {
-            'activedirectory': DSStatus.DISABLED.name,
-            'ldap': DSStatus.DISABLED.name,
-            'nis': DSStatus.DISABLED.name
-        }
+        self.logger.debug("XXX: new %s", new)
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/directoryservices.py:168]: leftover debugs here and below

File: src/middlewared/middlewared/plugins/directoryservices.py [hunk 3]
```diff
@@ -3,14 +3,21 @@
...
 from middlewared.schema import accepts
 from middlewared.service import Service, private
 from middlewared.plugins.smb import SMBCmd, SMBPath
-from middlewared.utils import run
+from middlewared.utils import filter_list, osc, run
 from samba.dcerpc.messaging import MSG_WINBIND_OFFLINE, MSG_WINBIND_ONLINE
 
+OS_TYPE_FREEBSD = 0x01
+OS_TYPE_LINUX = 0x02
+OS_FLAG = int(osc.IS_FREEBSD) + (int(osc.IS_LINUX) << 1)
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/directoryservices.py:19]: This is like C in Python which doesnt make the most sense as we are not hurting for saving a few bytes.
Mostly just noting the fact, no need to change :P

File: src/middlewared/middlewared/etc_files/master.passwd [hunk 4]
```diff
@@ -17,7 +17,7 @@ else:
 %>\
 ${user['username']}:${passwd}:${user['uid']}:${user['group']['bsdgrp_gid']}:${freebsd_fields}${user['full_name']}:${user['home']}:${user['shell']}
 % endfor
-% if middleware.call_sync('nis.config')['enable']:
+% if IS_FREEBSD and middleware.call_sync('nis.config')['enable']:
```
Comment (themylogin) [inline src/middlewared/middlewared/etc_files/master.passwd:20]: This is a FreeBSD-only file

File: src/middlewared/middlewared/etc_files/nsswitch.conf [hunk 5]
```diff
@@ -12,7 +12,7 @@
 
         ad_enabled = safe_call('activedirectory.config')['enable']
         ldap_enabled = safe_call('ldap.config')['enable']
-        nis_enabled = safe_call('nis.config')['enable']
+        nis_enabled = safe_call('nis.config')['enable'] if IS_FREEBSD else False
```
Comment (themylogin) [inline src/middlewared/middlewared/etc_files/nsswitch.conf:15]: No need to change: `IS_FREEBSD and safe_call('nis.config')['enable']` like did below

File: src/middlewared/middlewared/plugins/directoryservices.py (hunk 3 above)
Comment (themylogin) [reply src/middlewared/middlewared/plugins/directoryservices.py:19]: That's one WEIRD line of code

**Comment by william-gr** [comment]
@anodos325 this has been hanging around for around 6 months. Time to get this resolved for good or close the PR and retarget the ticket, please.
